backend/resolvers.js

```diff
@@ -0,0 +1,18 @@
+export const resolvers = {
+    Query: {
+        posts: (parent, args, context) => context.dataSources.db.posts,
+        users: (parent, args, context) => context.dataSources.db.users,
+    },
+    User: {
+        posts: (parent, args, context, info) =>
+            context.dataSources.db.posts.filter((post) => post.author.name === parent.name)
+    },
+    Post: {
+        votes: (parent, args, context) => context.dataSources.db.getVotes(parent.id)
```

Comment:
`parent` has your votes already here!
```suggestion
        votes: (parent, args, context) => parent.votes.length
```